softwareComponents/torqueComputation/src/joint.cpp

```diff
@@ -0,0 +1,212 @@
+            // No equation to add
+            return std::nullopt;
+        }
+
+        mat momentLhs(3, variableManager.getVariablesCount(), fill::zeros);
+
+        for (Joint* neighbor : neighbors) {
+            int ownTorqueVars = variableManager.getMomentStartIndexForEdge(id, neighbor->id);
+
+            for (int i = 0; i < 3; i++) {
+                momentLhs.col(ownTorqueVars + i) += moments.row(i).t();
+            }
+        }
+
+        return momentLhs;
+    }
+
+    mat Joint::createMomentEdgeLhs(const Joint& neighbor, const VariableManager& variableManager) const {
+        mat  edgeMomentLhs(3, variableManager.getVariablesCount(), fill::zeros);
+
+        int ownTorqueVars = variableManager.getMomentStartIndexForEdge(id, neighbor.id);
+        int neighborTorqueVars = variableManager.getMomentStartIndexForEdge(neighbor.id, id);
+
+        mat  ownMomentsProjected = getMomentsProjectedOnEdge(neighbor);
+        mat  neighborMomentsProjected = neighbor.getMomentsProjectedOnEdge(*this);
+
+        for (int i = 0; i < 3; i++) {
+             edgeMomentLhs.col(ownTorqueVars + i) +=  ownMomentsProjected.row(i).t();
+             edgeMomentLhs.col(neighborTorqueVars + i) -=  neighborMomentsProjected.row(i).t();
+        }
+
+        return  edgeMomentLhs;
+    }
+
+    mat33 Joint::getMomentForces(const Joint& neighbor) const {
+        std::vector<int>  canonicNeighborsIds = getCanonicNeighborsIds();
+        if (std::find(
+             canonicNeighborsIds.begin(),
+             canonicNeighborsIds.end(), 
+            neighbor.id) ==  canonicNeighborsIds.end()
+        ) {
+            throw std::logic_error("Provided joint is not a neighbor of handled joint");
+        }
+
+        vec  neighborVector = neighbor.coors - coors;
+        mat  momentForces = projectMatrixOntoPlane(moments,  neighborVector) * getCrossDivisionMatrix(neighborVector);
+
+        return  momentForces;
+    }
+
+    mat33 Joint::getMomentsProjectedOnEdge(const Joint& neighbor) const {
+        std::vector<int>  canonicNeighborsIds = getCanonicNeighborsIds();
+        if (std::find(
+             canonicNeighborsIds.begin(),
+             canonicNeighborsIds.end(), 
+            neighbor.id) ==  canonicNeighborsIds.end()
+        ) {
+            throw std::logic_error("Provided joint is not a neighbor of handled joint");
+        }
+
+        vec  neighborVector = neighbor.coors - coors;
+        return projectMatrixOntoVector(moments,  neighborVector);
+    }
+
+    void Joint::appendNeighbor(Joint* neighbor) {
+        neighbors.push_back(neighbor);
+    }
+
+    void Joint::appendForce(arma::vec3 force) {
+        forces.push_back(force);   
+    }
+
+    Joint* Joint::getNeighbourById(int  neighborId) {
+        for (Joint* neighbor : neighbors) {
+            if (neighbor->id ==  neighborId) {
+                return neighbor;
+            }
+        }
+        return nullptr;
+    }
+
+    bool Joint::jointSanityCheck(std::string& msg) const {
```

Comment:
Kontrakt této funkce se mi zdá neohrabaný. Jediné místo, kde se používá je pro vyhození vyjímky. Tak proč rovnou neháže?